src/django_components/commands/ext_run.py

```diff
@@ -0,0 +1,113 @@
+from typing import Any, List, Optional, Type
+
+from django_components.extension import extensions
+from django_components.util.command import ComponentCommand
+
+
+# Scope the extension-specific commands to the extension name, so users call these commands like:
+# `python manage.py components ext run <extension> <command>`
+#
+# We achieve that by creating temporary `ComponentCommand` subclasses for each extension. E.g.
+# ```python
+# class ExtCommand(ComponentCommand):
+#     name = "my_ext"
+#     help = "Run commands added by the 'my_ext' extension."
+#     subcommands = [
+#         ...commands
+#     ]
+# ```
+def _gen_subcommands() -> List[Type[ComponentCommand]]:
+    commands: List[Type[ComponentCommand]] = []
+    for extension in extensions.extensions:
+        ExtCommand = type(
+            "ExtCommand",
+            (ComponentCommand,),
+            {
+                "name": extension.name,
+                "help": f"Run commands added by the '{extension.name}' extension.",
+                "subcommands": extension.commands,
+            },
+        )
+
+        commands.append(ExtCommand)
+
+    return commands
+
+
+# This descriptor generates the list of subcommands of the `components ext run` command dynamically when accessed.
+# This is because this list depends on the settings and extensions. In tests, the settings and available extensions
+# may change between each test case. So we have to ensure we access the latest settings when accessing this property.
+#
+# NOTE: This is possible, because Django sets up the project and settings BEFORE the commands are loaded.
+class SubcommandsDescriptor:
+    def __get__(self, obj: Any, objtype: Optional[Type] = None) -> List[Type[ComponentCommand]]:
+        # This will be called when accessing ExtRunCommand.subcommands
+        # or instance.subcommands
+        return _gen_subcommands()
+
+
+class ExtRunCommand(ComponentCommand):
```

Comment:
Definition for `python manage.py components ext run`. This is the command that one can use to run commands defined by extensions. E.g. the idea is that the storybook integration could use something like:

`python manage.py components ext storybook generate`